src/accel/TrackingManagerOffload.cc

```diff
@@ -0,0 +1,141 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2023 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file accel/TrackingManagerOffload.cc
+//---------------------------------------------------------------------------//
+#include "TrackingManagerOffload.hh"
+
+#include <G4ProcessManager.hh>
+#include <G4ProcessVector.hh>
+#include <G4Track.hh>
+
+#include "corecel/Assert.hh"
+
+#include "ExceptionConverter.hh"
+#include "LocalTransporter.hh"
+#include "SharedParams.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Construct a tracking manager with data needed to offload to Celeritas
+ *
+ * TODO: Clarify thread-locality. Construction/addition to \c
+ * G4ParticleDefinition appears to take place on the master thread, typically
+ * in the ConstructProcess method, but the tracking manager pointer is part of
+ * the split-class data for the particle. It's observed that different threads
+ * have distinct pointers to a LocalTransporter instance, and that these match
+ * those of the global thread-local instances in test problems.
+ */
+TrackingManagerOffload::TrackingManagerOffload(SharedParams const* params,
+                                               LocalTransporter* local)
+    : params_(params), transport_(local)
+{
+    CELER_EXPECT(params);
+    CELER_EXPECT(local);
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Build physics tables for this particle
+ *
+ * Messaged by the \c G4ParticleDefinition who stores us whenever cross-section
+ * tables have to be rebuilt (i.e. if new materials have been defined). An
+ * override is needed for Celeritas as it uses the particle's process manager
+ * and tables to initialize its own physics data for the particle, and this is
+ * disabled when a custom tracking manager is used. Note that this also means
+ * we could have filters in HandOverOneTrack to hand back the track to the
+ * general G4TrackingManager if matching a predicate(s).
+ *
+ * The implementation follows that in \c G4VUserPhysicsList::BuildPhysicsTable
+ * , see also Geant4 Extended Example runAndEvent/RE07.
+ */
+void TrackingManagerOffload::BuildPhysicsTable(G4ParticleDefinition const& part)
+{
+    G4ProcessManager* pManager = part.GetProcessManager();
+    G4ProcessManager* pManagerShadow = part.GetMasterProcessManager();
+
+    G4ProcessVector* pVector = pManager->GetProcessList();
+    for (std::size_t j = 0; j < pVector->size(); ++j)
```

Comment:
```suggestion
    CELER_ASSERT(pManager && pManager->GetProcessList());
    for (G4VProcess* process : *pManager->GetProcessList())
```
and below `(*pVector)[j]` -> `process`

Reply:
Despite its name, `G4ProcessVector` doesn't provide an interface compatible with, or trivially adaptable to, range-based for ☹️! We could write the loop as is done in `GeantImporter`:

```cpp
    CELER_ASSERT(pManager);
    G4ProcessVector* pVector = pManager->GetProcessList();
    CELER_ASSERT(pVector);
    for(auto j : range(pVector->size())
    {
        G4VProcess* p = *pVector[j];
        ....
```
?

Reply:
Implemented as above, which I think is the best that can be done without adding an iterator adaptor for `G4ProcessVector` (will add a Geant4 issue to review collection interfaces for "proper" behaviour!).

Reply:
Ha, sounds good. Cheers!